apps/e2e/ag-grid-storybook-e2e/cypress.config.ts

```diff
@@ -1,21 +1,138 @@
+import * as os from 'os';
+import * as path from 'path';
 
-const cypressJsonConfig = {
-  fileServerFolder: '.',
-  fixturesFolder: './src/fixtures',
-  video: false,
-  videosFolder: '../../../dist/cypress/apps/e2e/ag-grid-storybook-e2e/videos',
-  screenshotsFolder:
-    '../../../dist/cypress/apps/e2e/ag-grid-storybook-e2e/screenshots',
-  chromeWebSecurity: false,
-  specPattern: 'src/e2e/**/*.cy.{js,jsx,ts,tsx}',
-  supportFile: 'src/support/e2e.ts',
-};
 export default defineConfig({
   e2e: {
     ...nxE2EPreset(__dirname),
-    ...cypressJsonConfig,
+    video: false,
+    setupNodeEvents: (on, config) => {
+      // Use a large window size so screenshots fit within the viewport.
+      on('before:browser:launch', (browser, launchOptions) => {
+        const width = 1920;
+        const height = 1600;
+
+        if (browser.name === 'chrome' && browser.isHeadless) {
+          launchOptions.args.push(`--window-size=${width},${height}`);
+          launchOptions.args.push('--force-device-scale-factor=2');
+          launchOptions.args.push('--no-sandbox');
+          launchOptions.args.push('--disable-setuid-sandbox');
+        }
+
+        if (browser.name === 'electron' && browser.isHeadless) {
+          launchOptions.preferences.width = width;
+          launchOptions.preferences.height = height;
+          launchOptions.preferences.deviceScaleFactor = 2;
+        }
+
+        if (browser.name === 'firefox' && browser.isHeadless) {
+          launchOptions.args.push(`--window-size=${width},${height}`);
+        }
+
+        return launchOptions;
+      });
+
+      const snapshotDotYml = config.screenshotsFolder + '/percy-snapshots.yml';
+      on('before:run', async () => {
+        if (!config.screenshotsFolder) {
+          return;
+        }
+        fs.mkdirSync(config.screenshotsFolder, { recursive: true });
+        fs.writeFileSync(snapshotDotYml, `snapshots:${os.EOL}`);
+      });
+      on('after:screenshot', (details: Cypress.ScreenshotDetails) => {
+        if (!config.screenshotsFolder) {
+          return;
+        }
+        const htmlFile = details.path.replace('.png', '.html');
+        const relativePath = path.relative(config.screenshotsFolder, htmlFile);
+        const imageUrl = path.basename(details.path);
+        const html = `
+        <!doctype html>
+        <html lang="en">
+          <head>
+            <meta charset="utf-8">
+            <title>${details.name}</title>
+            <style>
+              *, *::before, *::after { margin: 0; padding: 0; font-size: 0; }
+              html, body { width: 100%; }
+              img { max-width: 100%; }
+            </style>
+          </head>
+          <body>
+            <img
+              alt="${details.name}"
+              src="${imageUrl}"
+              width="${details.dimensions.width}px"
+              style="aspect-ratio: ${details.dimensions.width}/${details.dimensions.height};"
+            />
+          </body>
+        </html>`;
```

Comment:
This HTML wrapper mimics what `npx percy upload` does but allows controlling the snapshot name.